File: Problems/Distribute Elements Into Two Arrays I.py
# Loop through index and distribute based on largest values
class Solution:
    def resultArray(self, nums):
        len_nums = len(nums)
        
        arr1 = [nums[0]]
        arr2 = [nums[1]]

        i = 2
        while (i < len_nums):
            if (arr1[-1] > arr2[-1]):
                arr1.append(nums[i])
            else:
                arr2.append(nums[i])
            
            i += 1

        return arr1 + arr2

File: Problems/Distribute Elements Into Two Arrays I.cpp
class Solution {
public:
    vector<int> resultArray(vector<int>& nums) {
        vector<int> arr1;
        vector<int> arr2;

        int n = nums.size();
        arr1.push_back(nums[0]);
        arr2.push_back(nums[1]);

        for (int i = 2; i < n; i++) {
            if (*(--arr1.end()) >= *(--arr2.end())) {
                arr1.push_back(nums[i]);
            } else {
                arr2.push_back(nums[i]);
            }
        }
        arr1.insert(arr1.end(), arr2.begin(), arr2.end());
        return arr1;
    }
};
